testing/examples/state_tests.rs

```diff
@@ -89,13 +89,26 @@ async fn main() {
 	let _keys = api.get_keys(storage_key, None).unwrap().unwrap();
 	let _constants: Balance = api.get_constant("Balances", "ExistentialDeposit").unwrap();
 
-	let max_keys = 3;
+	let max_keys = 2003;
+	let result = api.get_storage_keys_paged_limited(
+		Some(storage_key_prefix.clone()),
+		max_keys.clone(),
+		None,
+		None,
+	);
+	assert!(result.is_err());
+	assert!(format!("{result:?}").contains("count exceeds maximum value"));
+
 	let storage_keys = api
 		.get_storage_keys_paged(Some(storage_key_prefix), max_keys, None, None)
 		.unwrap();
-	assert_eq!(storage_keys.len() as u32, max_keys);
+	assert_eq!(storage_keys.len() as u32, 13);
```

Comment:
Hmm.. that's too bad. So we basically don't fully test our function loop currently. So I suppose we should introduce a  unit test. But for this we need a mock. Could be done in a separate issue. What do you think?

Reply:
Add issue: https://github.com/scs/substrate-api-client/issues/605